app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt

```diff
@@ -44,6 +46,26 @@ class AddTransactionsAdapter(
         }
     }
 
+    fun setDate(position: Int, day: Int, month: Int, year: Int) {
+        val transaction = adapterData[position]
+                as? AddTransactionItem.Transaction
+
+        val rightNow: Calendar = Calendar.getInstance()
+        rightNow.set(Calendar.MILLISECOND, 0)
+        rightNow.set(Calendar.SECOND, 0)
+        rightNow.set(Calendar.MINUTE, 0)
+        rightNow.set(Calendar.HOUR, 0)
+        rightNow.set(Calendar.DAY_OF_MONTH, day)
+        rightNow.set(Calendar.MONTH, month)
+        rightNow.set(Calendar.YEAR, year)
+
+        transaction?.date = rightNow.timeInMillis
+
+        transaction?.let {
+            notifyItemChanged(adapterData.indexOf(it))
```

Comment:
Actually you don't have to do it like this - you have the `position` from the argument so you can just do:

`notifyItemChanged(position)`